auth/service.go

```diff
@@ -151,6 +173,21 @@ func (svc service) RetrieveKey(ctx context.Context, token, id string) (Key, erro
 }
 
 func (svc service) Identify(ctx context.Context, token string) (Key, error) {
+	if strings.HasPrefix(token, patPrefix+patSecretSeparator) {
+		pat, err := svc.IdentifyPAT(ctx, token)
+		if err != nil {
+			return Key{}, err
+		}
+		return Key{
+			ID:        pat.ID,
+			Type:      PersonalAccessToken,
+			Subject:   pat.User,
+			User:      pat.User,
+			IssuedAt:  pat.IssuedAt,
+			ExpiresAt: pat.ExpiresAt,
+		}, nil
+	}
```

Comment:
Move this logic to `Authenticate` function in file `pkg/authn/authsvc/authn.go`

Lets have seperate RPC for IdentifyPAT. 

The `Authenticate` function in file `pkg/authn/authsvc/authn.go` will call IdentifyPAT base on token prefix